File: ruby/easy/longest_common_prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  res, len = "", 0
  return res if(strs.length == 0)
  len = strs[0].length
  strs.each do |str|
    len = [len, str.length].min
  end
  
  for i in 0..len
    strs.each do |str|
      return res if(str[i] != strs[0][i])
    end
    res += strs[0][i].to_s
  end
  return res
end

File: javascript/easy/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let i = 0, len = Number.MAX_SAFE_INTEGER, move = true, res = "";
    if(strs.length == 0) return res;

    for(let str of strs) {
        len = Math.min(len, str.length);
    }

    while(i < len) {
        for(let j = 0; j < strs.length; j++) {
            if(strs[j][i] !== strs[0][i]) {
                return res;
            }
        }
        res += strs[0][i];
        i++;
    }
    return res;
};